zerver/actions/custom_profile_fields.py

```diff
@@ -141,6 +191,30 @@ def try_update_realm_custom_profile_field(
         if field_data is not None or field.field_data == "":
             field.field_data = orjson.dumps(field_data or {}).decode()
     field.save()
+
+    RealmAuditLog.objects.create(
+        realm=realm,
+        acting_user=acting_user,
+        event_type=AuditLogEventType.REALM_CUSTOM_PROFILE_FIELD_UPDATED,
+        event_time=timezone_now(),
+        extra_data={
+            "field_id": field.id,
+            RealmAuditLog.OLD_VALUE: {
+                "name": old_name,
+                "hint": old_hint,
+                "required": old_required,
+                "editable_by_user": old_editable_by_user,
+                "display_in_profile_summary": old_display_in_profile_summary,
+            },
+            RealmAuditLog.NEW_VALUE: {
+                "name": field.name,
+                "hint": field.hint,
+                "required": field.required,
+                "editable_by_user": field.editable_by_user,
+                "display_in_profile_summary": field.display_in_profile_summary,
```

Comment:
Other way here can be to add a different entry for each property being changed by adding a `property` field to `extra_data` so that we do not need to store the values that haven't been changed. But am not sure what would be better